### PR TITLE
[Merged by Bors] - More explicit help how to cycle the cameras

### PR DESCRIPTION
**Pull request by fvacek**
# Objective

Scene viewer example has switch camera keys defined, but only one camera was instantiated on the scene.

## Solution

More explicit help how to cycle the cameras, explaining that more cameras must be present in loaded scene.

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Usually the "switch camera" button is actually used for scenes that have included cameras :) I think the correct fix is actually just to clarify that in the docs of this example.

**Comment by fvacek** [comment]
Current help says that `C` key switches the cameras, so we should delete one line in help or add one more camera. I've chosen the second option because it is much more fun with second camera.
```
[~/p/bevy]$ cargo run --example scene_viewer                                                                                [scene-viewer-second-camera-added]

    Finished dev [unoptimized + debuginfo] target(s) in 0.44s
     Running `target/debug/examples/scene_viewer`

Controls:
    MOUSE       - Move camera orientation
    LClick/M    - Enable mouse movement
    WSAD        - forward/back/strafe left/right
    LShift      - 'run'
    E           - up
    Q           - down
    L           - animate light direction
    U           - toggle shadows
    C           - cycle through cameras
    5/6         - decrease/increase shadow projection width
    7/8         - decrease/increase shadow projection height
    9/0         - decrease/increase shadow projection near/far

    Space       - Play/Pause animation
    Enter       - Cycle through animations
```

File: examples/tools/scene_viewer.rs [hunk 1]
```diff
@@ -237,24 +237,41 @@ fn setup_scene_after_load(
...
-                    is_active: false,
+                    is_active: true,
                     ..default()
                 },
                 ..default()
-            },
-            CameraController::default(),
-        ));
+            })
+            .insert(CameraController::default());
```
Comment (DJMcNab) [inline examples/tools/scene_viewer.rs:256]: Why have you changed this into the unidiomatic form?

File: examples/tools/scene_viewer.rs [hunk 2]
```diff
@@ -457,80 +474,90 @@ fn camera_controller(
...
-        if !options.enabled {
-            return;
-        }
+    let active_camera_id = if let Some(camera) = camera_tracker.active_camera() {
+        camera.id()
+    } else {
+        return;
+    };
+
+    query.for_each_mut(|(entity, mut transform, mut options)| {
```
Comment (DJMcNab) [inline examples/tools/scene_viewer.rs:488]: Why not use `query.get_mut(entity)` here?

**Comment by rparrett** [comment]
The `C` currently button works as expected as long as there is at least one camera in the loaded scene in addition to the provided camera controller.

So I don't personally understand the need to add additional camera controllers or alter the usage text.

Screen Shot 2022-10-04 at 4 33 04 PM

thing

**Comment by fvacek** [comment]
Well, now I guess I understand what happened to me as a beginner. I just run an example, pushed C according to help and nothing happened. I didn't realize that there might be more cameras loaded together with scene, unfortunately the default scene does not contain one, so `C` key behavior was confusing to me. 

There are possible fixes in my mind:
1. change the help text to something like `cycle through cameras, if loaded scene contains any`
2. add camera to scene, which is loaded by default
3. Add info line with explanation that loaded scene does not contain any camera, so `C` will not work as expected.

Thank you for explanation.

**Comment by alice-i-cecile** [comment]
I think I'd like both 1 and 3 here :)

**Comment by rparrett** [comment]
It would be nice to clarify that camera controls do not work while viewing one of the loaded cameras as well.

Something like
```
    C           - cycle through the camera controller and any cameras loaded from the scene
```
might cover all the bases.

**Comment by fvacek** [comment]
Problem is, that even if I add more cameras to the scene, I'm not able to cycle through them using `C`.
[brick.zip](https://github.com/bevyengine/bevy/files/9717384/brick.zip)

**Comment by rparrett** [comment]
That scene doesn't have any cameras. If it was authored in Blender, did you check the "include cameras" option when exporting the glb?

Screen Shot 2022-10-05 at 9 40 16 AM

**Comment by fvacek** [comment]
No, I didn't, thank you very much for hint.

**Comment by fvacek** [comment]
So, only thing what might be missing for now is to change the help text to 
```
    C           - cycle through the camera controller and any cameras loaded from the scene
```
should I make a new PR with better title or can we use this one?

**Comment by alice-i-cecile** [comment]
Just edit this PR's title and description.

**Review by alice-i-cecile — APPROVED** [review]

**Review by rparrett — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
Awesome, thanks for talking through this so patiently.

bors r+

**Pull request closed without merging**